### ISSUE TITLE
Missing Realm Constructor & Manually link when RN CLI uses autolinking for native dependencies

### ISSUE THREAD

**Issue by toto104** [realm/realm-js#2588]
## Goals
Be able to install Realm in my react native project

## Expected Results
successful install

## Actual Results
I'm getting **Error: Missing Realm constructor. Did you run "react-native link real"?...**

## Steps to Reproduce
1. I create a new project via terminal: `react-native init test01`
2. `cd test01`
3. `react-native run-ios`
4. The simulator opens the app perfectly. Then on `App.js` I delete all the existing code and write a very simple code to display a HelloWorld. I saved the files and reload the simulator.
5. Then on terminal I procede to install Realm: `npm install --save realm`. successfully installed.
6. `react-native link realm`
7. Dependencies successfully linked.
8. I go to the simulator, click **reload** and runs perfect.
9. So then I go to `App.js` to add the Realm part of the code:
10. I Reload the simulator and I get the error.

I also get an error on terminal when I try to re-run the app `react-native run-ios`:
**error React Native CLI uses autolinking for native dependencies, but the following modules are linked manually - realm (to unlink run "react-native unlink real")**

## Code Sample
```
import React from 'react';
import {SafeAreaView, View, Text} from 'react-native';

const Realm = require('realm');

export default class App extends React.Component {
    constructor(props) {
    super(props);
    this.state = { realm: null };
  }

  componentWillMount() {
    Realm.open({
      schema: [{name: 'Dog', properties: {name: 'string'}}]
    }).then(realm => {
      realm.write(() => {
        realm.create('Dog', {name: 'Rex'});
      });
      this.setState({ realm });
    });
  }

  render() {
    return (<SafeAreaView><View><Text>HelloWorld</Text></View></SafeAreaView>)    
  }
}
```

## Version of Realm and Tooling
- Realm versions tried: @3.3.0 / @3.2.0 / @3.1.0
- react-native-cli: 2.0.1
- react-native: 0.61.3
- Node: v10.17.0
- OS: macOS v10.15.1

**Comment by kraenhansen** [comment]
We're working on a fix for autolinking (see https://github.com/realm/realm-js/pull/2586), please tryout the PR following the instructions here: https://github.com/realm/realm-js/issues/2460#issuecomment-548808577.

**Comment by toto104** [comment]
> We're working on a fix for autolinking (see #2586), please tryout the PR following the instructions here: [#2460 (comment)](https://github.com/realm/realm-js/issues/2460#issuecomment-548808577).

@kraenhansen Thanks for your reply, it works! Fantastic! I followed all the steps you mention on issue 2460. One thing I notice is that It took a while (around 10m) to install RealmJS (3.3.0) when I run `pod install`.

I would like to know more about this issue. Do you know why the Realm installation has to go this way? and not as it is mention on the Realm page? like this:
- `npm install --save realm`
- `react native link real`

Thanks again!

**Comment by kraenhansen** [comment]
@toto104 the `kh/podspec` branch will be deleted as soon as the change is merged into master. ETA is a couple of days from now. If I were you I would definitely wait for the next official release of Realm JS.
